packages/bridge/src/memory.ts

```diff
@@ -83,3 +85,26 @@ export function logMemoryError(err: any) {
       break;
   }
 }
+
+export function registerGarbageCollectionObserver() {
+  const obs = new PerformanceObserver(items => {
+    items
+      .getEntries()
+      .filter(
+        item =>
+          (item.detail as NodeGCPerformanceDetail).kind === constants.NODE_PERFORMANCE_GC_MAJOR,
```

Comment:
Should we use optional chaining here as a precaution since `detail` can be nullable (`NodeGCPerformanceDetail | unknown | undefined`)? Or are you certain that it will never be null?

```suggestion
          (item.detail as NodeGCPerformanceDetail)?.kind === constants.NODE_PERFORMANCE_GC_MAJOR,
```